packages/plugin-core/src/test/suite-integ/AddAndCommitCommand.test.ts

```diff
@@ -0,0 +1,215 @@
+import { ConfigUtils, NoteUtils } from "@dendronhq/common-all";
+import { tmpDir } from "@dendronhq/common-server";
+import {
+  Git,
+  SyncActionStatus,
+  WorkspaceUtils,
+} from "@dendronhq/engine-server";
+import { GitTestUtils } from "@dendronhq/engine-test-utils";
+import _ from "lodash";
+import { expect } from "../testUtilsv2";
+import { describeSingleWS } from "../testUtilsV3";
+import fs from "fs-extra";
+import { ExtensionProvider } from "../../ExtensionProvider";
+import { AddAndCommit } from "../../commands/AddAndCommit";
+import { NoteTestUtilsV4 } from "@dendronhq/common-test-utils";
+
+const TIMEOUT = 60 * 1000 * 5;
+
+suite("GIVEN Workspace Add And Commit command is run", function () {
+  this.timeout(TIMEOUT);
+  describeSingleWS(
+    "WHEN there are no changes",
+    {
+      modConfigCb: (config) => {
+        ConfigUtils.setWorkspaceProp(config, "workspaceVaultSyncMode", "sync");
+        return config;
+      },
+    },
+    () => {
+      test("THEN Dendron skips committing", async () => {
+        const { wsRoot } = ExtensionProvider.getDWorkspace();
+        const remoteDir = tmpDir().name;
+        await GitTestUtils.createRepoForRemoteWorkspace(wsRoot, remoteDir);
+        // Add everything and push, so that there's no changes
+        const git = new Git({ localUrl: wsRoot });
+        await git.addAll();
+        await git.commit({ msg: "add all and commit" });
+        const out = await new AddAndCommit().execute();
+        const { committed, finalMessage } = out;
+        expect(WorkspaceUtils.getCountForStatusDone(committed)).toEqual(0);
+        expect(committed[0].status).toEqual(SyncActionStatus.NO_CHANGES);
+        expect(finalMessage).toEqual(
+          "Finished Commit. Skipped vault because it has no new changes. Committed 0 repo"
+        );
+      });
+    }
+  );
+
+  describeSingleWS(
+    "WHEN there is a merge conflict",
+    {
+      modConfigCb: (config) => {
+        ConfigUtils.setWorkspaceProp(config, "workspaceVaultSyncMode", "sync");
+        return config;
+      },
+    },
+    () => {
+      test("THEN Dendron skips committing files", async () => {
```

Comment:
can we make the test descriptions more specific? eg. `skips committing files AND shows merge conflict message`

even better if you can do verification across multiple test cases  (see below for example) but if its not trivial to setup, then just combine it in one test statement

```
test("THEN skip comitting files"

test("THEN show merge conflict message"
```